src/common/utils.py

```diff
@@ -1,21 +1,33 @@
-
-from typing import List, Dict, Tuple, Set, Union, Any, Type, Callable
+from __future__ import annotations
+from typing import List, Dict, Tuple, Set, Union, Any, Type, Callable, cast
```

Comment:
Suggest a docstring at the top of the file outlining its use and what code should go here.